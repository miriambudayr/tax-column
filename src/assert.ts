export function assert(v: any, why = ""): asserts v {
  if (!v) {
    const error = new Error(`Assertion Failed: ${why}`);
    error.name = "AssertionFailure";

    console.error("AssertionFailure", error);
    throw error;
  }
}
