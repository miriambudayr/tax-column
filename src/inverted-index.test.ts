import { InvertedIndex } from ".";
import dotenv from "dotenv";
import { assert } from "./assert";
dotenv.config();

describe("Inverted Index", () => {
  it("should index a file", () => {
    const fileName = "src/lorem-ipsum.txt";
    const invertedIndex = new InvertedIndex();
    invertedIndex.indexFile(fileName);

    const files = invertedIndex.search("metus");

    expect(files.size).toBeGreaterThan(0);

    if (files) {
      for (const file of files) {
        expect(file).toEqual(fileName);
      }
    }
  });

  it("should tokenize a directory", () => {
    const dirName = process.env.SAMPLE_DATA_DIR;
    assert(dirName, "SAMPLE_DATA_DIR variable not found in environment");
    const invertedIndex = new InvertedIndex();
    invertedIndex.recursivelyIndexDirectory(dirName);

    const files = invertedIndex.search("calendar");
    expect(Array.from(files!)).toMatchSnapshot();
  });
});
