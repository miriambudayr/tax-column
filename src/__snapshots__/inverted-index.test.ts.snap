// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Inverted Index should tokenize a directory 1`] = `
[
  "src/test_data/lorem-ipsum.txt",
]
`;
